service-base/src/main/java/org/eclipse/hono/service/AbstractProtocolAdapterBase.java

```diff
@@ -447,6 +449,37 @@ protected TrustOptions getServerTrustOptions() {
         return commandConnection.createCommandConsumer(tenantId, deviceId, messageConsumer, closeHandler);
     }
 
+    /**
+     * Validate that the <em>reply-to</em> and the <em>correlationId</em> of a command message are correctly built.
+     * If the validation is successful, a combined String <em>command request id</em> is returned.
+     * @param tenantId The tenant to be used for the validation.
+     * @param deviceId The device to be used for the validation.
+     * @param commandMessage The message containing a command.
+     * @return Optional An Optional with the combined String, or an empty Optional if the validation failed.
+     * @throws NullPointerException If any of the parameters are null.
+     */
+    protected final Optional<String> validateAndGenerateCommandRequestId(final String tenantId, final String deviceId, final Message commandMessage) {
+        Objects.requireNonNull(tenantId);
+        Objects.requireNonNull(deviceId);
+        Objects.requireNonNull(commandMessage);
+
+        // example of commandReplyId: control/DEFAULT_TENANT/4711/33fe70fd-5a2e-4095-83db-00101bf74a07
+        final String commandReplyId = commandMessage.getProperties().getReplyTo();
```

Comment:
commandMessage.getReplyTo() might be better.

Reply:
:+1: